src/compile/data.ts

```diff
@@ -479,12 +506,13 @@ export namespace nullFilter {
   }
 
   export function parseLayer(model: LayerModel) {
+    // note that we run this before source.parseLayer
     let nullFilterComponent = parse(model);
```

Comment:
We might not even need to call `parse` here because layers doesn't have its own field.  This is always an empty object.